fuzz/fuzz.c

```diff
@@ -0,0 +1,235 @@
+{
+#if ASS_FUZZMODE == FUZZMODE_STANDALONE
+    if (level > 6) return;
+    printf("libass: ");
+    vprintf(fmt, va);
+    printf("\n");
+#endif
+}
+
+static bool init(void)
+{
+    ass_library = ass_library_init();
+    if (!ass_library) {
+        printf("ass_library_init failed!\n");
+        return false;
+    }
+
+    ass_set_message_cb(ass_library, msg_callback, NULL);
+
+    ass_renderer = ass_renderer_init(ass_library);
+    if (!ass_renderer) {
+        ass_library_done(ass_library);
+        ass_library = NULL;
+        printf("ass_renderer_init failed!\n");
+        return false;
+    }
+
+    ass_set_fonts(ass_renderer, NULL, "sans-serif",
+                  ASS_FONTPROVIDER_AUTODETECT, NULL, 1);
+    ass_set_frame_size(ass_renderer, 854, 480);
+    ass_set_storage_size(ass_renderer, 854, 480);
+
+    return true;
+}
+
+static void consume_track(ASS_Renderer *renderer, ASS_Track *track)
+{
+    for (int n = 0; n < track->n_events; ++n) {
+        int change;
+        ASS_Event event = track->events[n];
+        ass_render_frame(ass_renderer, track, event.Start, &change);
+        if (event.Duration > 1) {
+            ass_render_frame(ass_renderer, track, event.Start + event.Duration/2, &change);
+            ass_render_frame(ass_renderer, track, event.Start + event.Duration-1, &change);
+        }
+    }
+}
+
+#if ASS_FUZZMODE == FUZZMODE_STANDALONE
+static ASS_Track *read_track_from_stdin(void)
+{
+    size_t smax = 4096;
+    char* buf = malloc(smax);
+    if (!buf)
+        goto error;
+    size_t s = 0;
+    ssize_t read_b = 0;
+    do {
+        // AFL++ docs recommend using raw file descriptors
+        // to avoid buffering issues with stdin
+        read_b = read(STDIN_FILENO, buf + s, smax - s);
+        s += read_b > 0 ? read_b : 0;
+        if (s == smax) {
+            size_t new_smax = smax > SIZE_MAX / 2 ? SIZE_MAX : smax * 2;
+            char* new_buf = realloc(buf, new_smax);
+            if (!new_buf || new_smax <= smax) {
+                free(new_buf ? new_buf : buf);
+                goto error;
+            }
+            smax = new_smax;
+            buf = new_buf;
+        }
+    } while (read_b > 0);
+    buf[s] = '\0';
+    ASS_Track *track = ass_read_memory(ass_library, buf, s, NULL);
+    free(buf);
+    return track;
+error:
+    printf("Input too large!\n");
+    return NULL;
+}
+
+int main(int argc, char *argv[])
+{
+    /* Default failure code of sanitisers is 1, unless
+     * changed via env (A|UB|..)SAN_OPTIONS=exitcode=21
+     * Except, LLVM's UBSAN always exits with 0 (unless using
+     * -fsanitize-undefined-trap-on-error which will SIGILL without an
+     * error report being printed), see https://reviews.llvm.org/D35085
+     */
+    enum {
+        FUZZ_OK = 0,
+        //SANITISER_FAIL = 1,
+        // Invalid parameters passed etc
+        FUZZ_BAD_USAGE = 2,
+        // Error before rendering starts
+        FUZZ_INIT_ERR = 0
```

Comment:
I set this to zero after some fuzzer-generated files were rejected by our parser as `TRACK_TYPE_UNKNOWN`, in the cases at hand because the fuzzer inserted a null-byte before the styles sections but iinm it could also happen if there was no Styles section at all. There might be some worth in keeping this enum for better communicating intent?

On a side note, iirc this differs from VSFilters which use the ScriptInfo `ScriptType` header and if at all reject based on that. Do we actually want to reject Styles-less files?